### PR TITLE
fix logical err of modify statistics

### PR DESCRIPTION
**Pull request by hanfei1991**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
fix logical error, when we create a replicated merge tree, alter a column and then execute modify statistics

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit 61fa4e7a476b3db31c22030470341b131501f3b6 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/68820/61fa4e7a476b3db31c22030470341b131501f3b6/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: src/Storages/StatisticsDescription.cpp [hunk 1]
```diff
@@ -115,8 +114,11 @@ void ColumnStatisticsDescription::merge(const ColumnStatisticsDescription & othe
 
 void ColumnStatisticsDescription::assign(const ColumnStatisticsDescription & other)
 {
+    /// If the statistics is empty, it's possible that we have not assign a column_name.
+    if (empty() && column_name == "")
+        column_name = other.column_name;
     if (other.column_name != column_name)
-        throw Exception(ErrorCodes::LOGICAL_ERROR, "Cannot assign statistics from column {} to {}", column_name, other.column_name);
+        throw Exception(ErrorCodes::LOGICAL_ERROR, "Cannot assign statistics from column {} to {}", other.column_name, column_name);
```
Comment (antaljanosbenjamin) [inline src/Storages/StatisticsDescription.cpp:121]: I think in `ColumnStatisticsDescription::fromAST` we always fill in `name` and `ColumnStatisticsDescription::fromAST` is called in `AlterCommand::apply`, so I am not sure how we end up with a `ColumnStatisticsDescription` without a filled in `name`.

Comment (antaljanosbenjamin) [reply src/Storages/StatisticsDescription.cpp:121]: Maybe just remove the `column_name` from `ColumnStatisticsDescription` and pass it in as a const reference to `assign` and `merge`? Actually merge is already doing it, so I think it should be easier. It is only used for logging as far as I see.

Comment (hanfei1991) [reply src/Storages/StatisticsDescription.cpp]: yes it's possible. I removed `column_name`

**Review by antaljanosbenjamin — APPROVED** [review]
Thank you @hanfei1991!

**Review by jsc0218 — APPROVED** [review]

**Pull request merged**